src/main/components/Browse/Filters.js

```diff
@@ -0,0 +1,262 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Select from "react-select";
+import makeAnimated from "react-select/lib/animated";
+import { withStyles } from "@material-ui/core/styles";
+
+const styles = theme => ({
+  clearButton: {
+    color: "#000000",
+    padding: "5px 10px",
+    marginTop: "20px",
+    borderRadius: "3px",
+    "&:focus": {
+      outline: "none"
+    },
+    "&:hover": {
+      backgroundColor: "#FFBD7C"
+    }
+  },
+  filterContainer: {
+    marginLeft: "10px",
+    marginTop: theme.spacing.unit * 3,
+    order: 1,
+    fontSize: 12
+  },
+  filterDiv: {
+    marginTop: "5px"
+  },
+  filterGroup: {
+    width: "200px"
+  },
+  span: {
+    color: "#000000"
+  }
+});
+
+class Filters extends Component {
+  constructor(props) {
+    super(props);
+    this.handleAnalysesChange = this.handleAnalysesChange.bind(this);
+    this.state = {
+      chosenFilters: {
+        title: null,
+        description: null,
+        jiraId: null,
+        libraryIds: null,
+        sampleIds: null,
+        project: null
+      },
+      analyses: this.props.analyses
+    };
+    this.filters = {
+      title: { label: "Title" },
+      description: { label: "Description" },
+      jiraId: { label: "Jira ID" },
+      libraryIds: { label: "Library ID" },
+      sampleIds: { label: "Sample ID" },
+      project: { label: "Project" }
+    };
+  }
+
+  handleAnalysesChange = analyses => this.props.onAnalysesChange(analyses);
+
+  /* 
+  Is the filter null? (ie. not present in the current selection)   
+    If so, then return true to the chain of ands, having no effect on the overall expression
+    If not, is the information being filtered an array in the analysis (ie. libraryIds or sampleIds)?
+      If so, does that array contain the chosen libraryId or sampleId?
+        If so, return true
+        If not, return false  
+      If not, then check if the filter in the selection matches the data in the analysis in question
+        If so, return true
+        If not, return false
+  */
+  isFilterWithinAnalysis = (analysis, filter, filters) =>
```

Comment:
This is fairly unreadable. I would suggest the following changes:

Use && and || (not sure if that's the proper Javascript syntax) instead of returning explicit true/false.
More importantly, break it down into smaller functions that have clear names

const isFilterLibraryOrSample
const isFilterContainedInArray
const isFiltersMatching

etc etc